crates/cairo-lint-core/src/lints/unwrap_used.rs

```diff
@@ -0,0 +1,38 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::{Expr, PathSegment};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{TypedSyntaxNode, Terminal};
+use cairo_lang_syntax::node::helpers::PathSegmentEx;
+use cairo_lang_semantic::resolve::AsSegments;
+
+
+/// Diagnostic message for usage of `unwrap()`
+pub const UNWRAP_USED: &str = "Use of unwrap() detected. Consider using '?' or 'expect()' instead.";
+
+/// Function to check for the usage of `unwrap()` in expressions.
+pub fn check_unwrap_used(
```

Comment:
use the `SemanticGroup` db to make sure to check for all the functions you can look at that it's a good example https://github.com/keep-starknet-strange/cairo-lint/pull/92